packages/@aws-cdk/aws-apigateway/lib/access-log.ts

```diff
@@ -0,0 +1,556 @@
+    return '$context.integrationStatus.';
+  }
+
+  /**
+   * The response latency in ms.
+   */
+  public static contextResponseLatency() {
+    return '$context.responseLatency';
+  }
+
+  /**
+   * The response payload length.
+   */
+  public static contextResponseLength() {
+    return '$context.responseLength';
+  }
+
+  /**
+   * The method response status.
+   */
+  public static contextStatus() {
+    return '$context.status';
+  }
+}
+
+/**
+ * A API Gateway custom access log format.
+ */
+export interface IAccessLogFormat {
+  /**
+   * A API Gateway custom access log format
+   */
+  readonly format: string;
+}
+
+/**
+ * jsonWithStandardFields property
+ */
+export interface IJsonWithStandardFieldProps {
+  /**
+   * If this flag is enabled, $context.identity.sourceIp will be output to the log
+   * @default true
+   */
+  ip?: boolean,
+  /**
+   * If this flag is enabled, $context.identity.caller will be output to the log
+   * @default true
+   */
+  caller?: boolean,
+  /**
+   * If this flag is enabled, $context.identity.user will be output to the log
+   * @default true
+   */
+  user?: boolean,
+  /**
+   * If this flag is enabled, $context.requestTime will be output to the log
+   * @default true
+   */
+  requestTime?: boolean,
+  /**
+   * If this flag is enabled, $context.httpMethod will be output to the log
+   * @default true
+   */
+  httpMethod?: boolean,
+  /**
+   * If this flag is enabled, $context.resourcePath will be output to the log
+   * @default true
+   */
+  resourcePath?: boolean,
+  /**
+   * If this flag is enabled, $context.status will be output to the log
+   * @default true
+   */
+  status?: boolean,
+  /**
+   * If this flag is enabled, $context.protocol will be output to the log
+   * @default true
+   */
+  protocol?: boolean,
+  /**
+   * If this flag is enabled, $context.responseLength will be output to the log
+   * @default true
+   */
+  responseLength?: boolean
+}
+
+/**
+ * factory methods for access log format.
+ */
+export class AccessLogFormat {
+  /**
+   * Generate Common Log Format.
+   */
+  public static clf(): IAccessLogFormat {
+    return {
+      format: `${AccessLogField.contextIdentitySourceIp()} ${AccessLogField.contextIdentityCaller()} ${AccessLogField.contextIdentityUser()} \
+[${AccessLogField.contextRequestTime()}] "${AccessLogField.contextHttpMethod()} ${AccessLogField.contextResourcePath()} ${AccessLogField.contextProtocol()}" \
+${AccessLogField.contextStatus()} ${AccessLogField.contextResponseLength()} ${AccessLogField.contextRequestId()}`
+    };
+  }
+
+  /**
+   * Generate standard json log format.
+   */
+  public static jsonWithStandardFields(
+    {
+      ip = true,
+      caller = true,
+      user = true,
+      requestTime = true,
+      httpMethod = true,
+      resourcePath= true,
+      status = true,
+      protocol = true,
+      responseLength = true
+    }: IJsonWithStandardFieldProps = {}): IAccessLogFormat {
```

Comment:
I'm not familiar with this pattern in typescript. Is this another way to set the default?

Normally, we do -

```ts
  public static jsonWithStandardFields(fields?: IJsonWithStandardFieldProps = {
      ip = true,
      caller = true,
      user = true,
      requestTime = true,
      httpMethod = true,
      resourcePath= true,
      status = true,
      protocol = true,
      responseLength = true
  }): IAccessLogFormat {
```

Are these the same?

Reply:
Yes.This is how the default values are set.